src/app/treasury/utils.ts

```diff
@@ -0,0 +1,35 @@
+import Big from '@/lib/big'
+import { formatNumberWithCommas } from '@/lib/utils'
+
+export interface AssetFormattingResult {
+  amount: string
+  symbol: string
+  fiatAmount?: string
+}
+
+/**
+ * Formats asset data for display in BalanceInfo components
+ */
+export function formatAssetData(
```

Comment:
Would it be good to write tests for this?

Reply:
done